src/bio/bio_config.c

```diff
@@ -425,49 +446,49 @@ load_bdev_subsystem_config(struct json_config_ctx *ctx, bool vmd_enabled,
 					 * to pass VMD address as the whitelist param.
 					 */
 					rc = traddr_to_vmd(traddr, traddr);
-					if (rc < 0) {
-						D_ERROR("Invalid traddr: %s\n", traddr);
-						rc = -DER_INVAL;
-						goto out;
+					if (rc != 0) {
+						D_ERROR("Invalid traddr %s (rc: %d)\n", traddr, rc);
+						goto free_traddr;
 					}
 
-					D_INFO("\t- VMD backing address reverted to '%s'\n",
-						traddr);
+					D_DEBUG(DB_MGMT, "\t- VMD backing address reverted to "
+						"'%s'\n", traddr);
 				}
 			}
 
 			rc = opts_add_pci_addr(opts, traddr);
-			if (rc < 0) {
+			if (rc != 0) {
 				D_ERROR("spdk env add pci: %d\n", rc);
-				goto out;
+				goto free_traddr;
 			}
-
 		}
-
 		key = spdk_json_next(key);
 	}
-out:
+
+free_traddr:
 	D_FREE(traddr);
-	free(cfg.method);
+free_method:
+	D_FREE(cfg.method);
+
+	if (rc > 0)
```

Comment:
Do not understand how this situation could happen ?

Reply:
L405 can return a positive rc, if L414 executes then the check on L473 is required

Reply:
Thanks for the clarification.